mongorestore/filepath.go

```diff
@@ -245,7 +245,7 @@ func (restore *MongoRestore) getInfoFromFile(filename string) (string, FileType,
 	}
 
 	// If the collection name is truncated, parse the full name from the metadata file.
-	if strings.Contains(collName, "%24") {
+	if strings.Contains(collName, "%24") && len(collName) == 238 {
```

Comment:
[TOOLS-2530](https://github.com/mongodb/mongo-tools/commit/105368810d1770e3b8c9f3a91b24586519210971) resulted in a mongorestore bug where collection names containing "$" were mishandled because of this truncation check; consequently, a few js tests failed. I've fixed this by adding `len(collName) == 238` to the check, which is true iff `collName` has been truncated.

Reply:
Good to know $ is allowed in collection name! @tfogo, seems this is contrary to the assumption in the design doc?

Reply:
Good to know $ is allowed in collection name! @tfogo, seems this is contrary to the assumption in the design doc?